api/handler/versioning.go

```diff
@@ -25,6 +26,8 @@ func (h *handler) PutBucketVersioningHandler(w http.ResponseWriter, r *http.Requ
 		return
 	}
 
+	h.log.Debug("target details", zap.String("cid", bktInfo.CID.String()))
```

Comment:
It seems like you call `n.log.Debug("target details",...)` after `h.logAndSendError(...)` everywhere in handler. I think it would be better to append `cid` and `oid` to `additionals` and pass it as a function parameter. And log `cid`, `oid` in `h.logAndSendError`.

Reply:
The idea was to make all methods do the logging as soon as cid/oid identified no matter of handler's result. And the `logAndSendError` gets called in case of error only.